_drafts/oryx/boilerplate.md

```diff
@@ -0,0 +1,139 @@
+
+## How we prevent boilerplate code
+
+In Oryx, we use a few tactics to prevent boilerplate code. These tactics include:
+
+1. Bootstrap the application from npm packages rather than source.
+2. Expose an function to setup the [app orchestrator](./app-orchestrator.md) conveniently.
+3. Provide [presets](./presets.md) for the standard application setup, including the feature sets and UI themes.
+4. Provide the source code in a [public repository](https://github.com/spryker/oryx) to enable developers to read the source code.
+5. Provide configurable component and business logic.
+6. Allow for customizations by [dependency injection](./dependency-injection.md).
+
+Most of the details around these tactics can be found elsewhere in the documentation. In the following sections you read more details on the boilerplate files that are needed.
+
+### NPM Packages
+
+One of the biggest sources of boilerplate code in any project is the application logic itself. In Oryx, we've separated out all the application logic into individual packages that are distributed on npm. These packages include components, business logic and integrations to the Spryker APIs.
+
+By separating out the application logic in this way, we've eliminated the boilerplate code from your projects. Instead of writing code to handle basic functionality, you can simply install the appropriate package and use it directly in your code. And because the packages are distributed on npm and published with [Semantic Versioning](https://semver.org/), you can be confident that upgrading to a new version of the framework won't break your code.
+
+### Using Presets
+
+Another source of boilerplate code in any project is the configuration required to get your application up and running. To simplify this process, we've introduced the concept of presets in Oryx.
+
+Presets provide configurations and data structures that are designed to get you up and running quickly without providing a lot of configuration yourself. Presets can be considered as so-called "demo applications", as they typically represent a demo application for a specific business model (e.g. "b2c demo shop"). By using presets, you can quickly configure your application without writing a lot of boilerplate code. This is particularly great when you first start your project or want to demo or test things out.
```

Comment:
Maybe a link to the `preset` folder in the `oryx` repo - so that devs can check it out?

Reply:
I'll make it a link to the preset package instead. We can consider making deeplinks from the preset package readme to the github code. But that's beyond this doc.